src/paas_charm/app.py

```diff
@@ -401,15 +401,21 @@ def gen_environment(self) -> dict[str, str]:  # noqa: too-complex
         if self._charm_state.peer_fqdns is not None:
             env[f"{prefix}PEER_FQDNS"] = self._charm_state.peer_fqdns
 
+        env.update(self._generate_integration_environments(prefix=self.integrations_prefix))
```

Comment:
refactor by wrapping integration environment updaters to `_generate_integration_environments` to facilitate testing in scoped chunks.